action.yaml

```diff
@@ -101,7 +107,7 @@ runs:
 
         echo "::set-output name=image::${image}"
 
-        if [[ "${{ inputs.dockerfile }}" != "" ]] || [[ "${{ inputs.context }}" != "" ]]; then
+        if [[ "${{ inputs.dockerfile }}" != "" ]] || [[ "${{ inputs.context }}" != "" ]] && [[ "${createGitTag}" == 'true' ]]; then
```

Comment:
It's getting messy. Let's rewind what this action can do 
* build from a given Cloudbuild config file 
* build from a given context 
* create git tag 


When building from a given context, the action is expecting to find a "version" file and a "release" file. 
When building from a given Cloudbuild config file, the action can't create a git tag. Hence why this test on line 110. 

If we want to support "createGitTag", I suggest we change the inputs to the following 


```
 # common
  cacheRepo: 'Docker repository to use as cache, required'
  image: 'full Docker image/tag to push, required if cacheRepo or config'
  buildArgs: 'multine string or build args with one line per arg as KEY=VALUE'
  createGitTag: 'should we create git tag with this release? default to true'

  # context build
  context: 'docker context, default to .'
  dockerfile: 'dockerfile, default to Dockerfile if context is set'
  
  # config build
  config: 'GCP Cloud Build configuration file location, required if image or cacheRepo'
  version: 'version to use for Docker Tag and Git Tag, required if  createGitTag (will override any Docker Tag set in 'image')" 
  releaseNote: 'note to be added to the Git Tag, can be multi line string, required if createGitTag

  ```
  
  What do you think ?

Reply:
my main motivation was to optionally disable git-tag-creating for context-build. 
Because I'm not sure that we want to pollute zenhub repo tags with anything else(like corgi-bot builds). Possibly naming not the best, and something like 'disableContextBuildReleaseNotes ' should be used. 

Not sure that we should implement releaseNote support for config-build, since we would like to move from it, so anybody want to have a release note -- just use the context-build option.

Reply:
It's a good idea to opt out creating git tag as indeed we don't want to create tag for every build we do. Essentially, I think we want to tag when feature branches are merged on master but that reflexion is not complete. We still need to discuss about it when reworking the Release Process. 
For the time being, here how you can test those changes 
- bump the minor version of that action (like 2.1) 
- edit this repo Github action to be able to pusblish you feature branch (to mimic what will happen when it get merged to staging) - so you will publish version 2.1 of this action 
- update the action for corgit in the dev repo to use that 2.1
- test and refine